### PR TITLE
BDBD-240 상품 사진 등록

### PR DESCRIPTION
**Pull request by minseonglove**
### 개요
글쓰기를 위한 상품 사진 등록 기능 추가

### 변경사항
* 상품 목록 api의 변경된 스펙 적용
* 상품 사진 등록 기능 추가
  * 글쓰기 화면에서 고양이 사진을 터치해 사진 선택 화면으로 입장 가능
* AlbumList~.kt
  * 사진 선택 화면 구현을 위한 클래스
* SelectedPictureListAdapter.kt
  * 선택한 사진들을 보여주기 위한 리사이클러 뷰 adapter
* DragAndDropCallback .kt
  * 선택한 사진들을 드래그 앤 드롭으로 순서 변경을 하기 위한 콜백 클래스

### 관련 지라 및 위키 링크
[BDBD-240](https://fake-developers.atlassian.net/jira/software/projects/BDBD/boards/10?selectedIssue=BDBD-240)

### 리뷰어에게 하고 싶은 말
[채신 릴리즈](https://github.com/FakeDevelopers/BidderBidder_AOS/releases/tag/2022-06-05-11-46)

### CODE REVIEW

**Comment by konempty** [comment]
> Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed](https://sonarcloud.io/dashboard?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35)
> 
> [Bug](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=CODE_SMELL) [5 Code Smells](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35) No Coverage information [](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&metric=new_duplicated_lines_density&view=list)

하하.. 코드스멜이 5개있네요 ㅋㅋㅋㅋ

File: app/src/main/kotlin/com/fakedevelopers/bidderbidder/ui/product_registration/album_list/AlbumListFragment.kt [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    private fun getPictures() {
+        val uri = MediaStore.Images.Media.EXTERNAL_CONTENT_URI
+        requireActivity().contentResolver.query(
+            uri,
+            null,
+            null,
+            null,
+            MediaStore.Images.ImageColumns.DATE_TAKEN + " DESC"
```
Comment (konempty) [inline app/src/main/kotlin/com/fakedevelopers/bidderbidder/ui/product_registration/album_list/AlbumListFragment.kt]: 흐음 imageId 라는 컬럼이 autoincreasement 였던것 같은데 그걸 쓰면 정렬하기 편하지 않은가요?

**Comment by konempty** [comment]
> Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed](https://sonarcloud.io/dashboard?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35)
> 
> [Bug](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=CODE_SMELL) [1 Code Smell](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35) No Coverage information [](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&metric=new_duplicated_lines_density&view=list)

헤헤 1개..

**Review by konempty — CHANGES_REQUESTED** [review]
크흐 고생하셨습니다! 따로 리뷰 드릴내용이 3개가 있네요
1. 사진 리소스중에 2MB짜리가 있는데 굳이 이렇게 클필요 있을까요? 이러면 apk파일도 무거워지고 설치도 오래 걸릴것 같아요 ㅋㅋㅋㅋ 사진 리소스는 용도에 맞게 리사이징 해서 넣어주세요!
2. 테스트를 위한 로그가 남아있는것 같은데 별로 좋아보이진 않더라구요 ㅋㅋㅋㅋ 에러 탐지나 꼭필요한 로그가 아니라면 테스트해보고 삭제해주세요!
3. 앱도 서버처럼 페이즈가 나눠져야 하지않을까? 라는 생각이 드네요 ㅋㅋㅋ 샌박앱과 리얼앱 2개로 서로다른 패키지명으로 빌드가 되어야 하고 샌박앱은 샌박 서버를, 리얼앱은 리얼서버를 바라보도록 해야하지않을까요?ㅋㅋㅋㅋㅋ 단, 지금은 서버에서 페이즈별 데이터가 혼합되어 쌓이고 있어서 의미는 없지만 추후에 서버에 관련기능이 생기면 고려해주세요!ㅋㅋㅋ

File: app/src/main/kotlin/com/fakedevelopers/bidderbidder/ui/product_registration/album_list/AlbumListFragment.kt (hunk 1 above)
Comment (minseonglove) [reply app/src/main/kotlin/com/fakedevelopers/bidderbidder/ui/product_registration/album_list/AlbumListFragment.kt]: imageId는 고려 안하고 단순히 시간 순으로 정렬하면 되겠구나라고 생각했읍니다.
imageId로 정렬해보고 시간 순으로 정렬한 것과 결과가 같다면 imageId로 진행하겠읍니다

**Comment by minseonglove** [comment]
> 크흐 고생하셨습니다! 따로 리뷰 드릴내용이 3개가 있네요
> 
> 1. 사진 리소스중에 2MB짜리가 있는데 굳이 이렇게 클필요 있을까요? 이러면 apk파일도 무거워지고 설치도 오래 걸릴것 같아요 ㅋㅋㅋㅋ 사진 리소스는 용도에 맞게 리사이징 해서 넣어주세요!
> 2. 테스트를 위한 로그가 남아있는것 같은데 별로 좋아보이진 않더라구요 ㅋㅋㅋㅋ 에러 탐지나 꼭필요한 로그가 아니라면 테스트해보고 삭제해주세요!
> 3. 앱도 서버처럼 페이즈가 나눠져야 하지않을까? 라는 생각이 드네요 ㅋㅋㅋ 샌박앱과 리얼앱 2개로 서로다른 패키지명으로 빌드가 되어야 하고 샌박앱은 샌박 서버를, 리얼앱은 리얼서버를 바라보도록 해야하지않을까요?ㅋㅋㅋㅋㅋ 단, 지금은 서버에서 페이즈별 데이터가 혼합되어 쌓이고 있어서 의미는 없지만 추후에 서버에 관련기능이 생기면 고려해주세요!ㅋㅋㅋ

1. 드자인 적용 전 임시로 넣는 야옹이 사진이라 용량에 신경을 안썼습니다. 리사이징된 야옹이 사진으로 바꾸겠습니다.
2. 다 색출해서 없애겠습니다.
3. 지당합니다. 요번 PR 이후에 따로 작업을 파서 나누도록 하겠읍니다.

**Comment by minseonglove** [comment]
> > Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed](https://sonarcloud.io/dashboard?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35)
> > [Bug](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=CODE_SMELL) [1 Code Smell](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&resolved=false&types=CODE_SMELL)
> > [No Coverage information](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35) No Coverage information [](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_AOS&pullRequest=35&metric=new_duplicated_lines_density&view=list)
> 
> 헤헤 1개..

메소드 하나 안쓰고 있어서 나오는 경곤데  이거 다음 작업에 바로 사용할거라 이거는 봐주십시오

File: app/src/main/kotlin/com/fakedevelopers/bidderbidder/ui/product_registration/DragAndDropCallback.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    override fun onSelectedChanged(viewHolder: RecyclerView.ViewHolder?, actionState: Int) {
+        super.onSelectedChanged(viewHolder, actionState)
+        if (isMoved) {
+            isMoved = false
+            adapter.changeMoveEvent()
+        }
+    }
+
+    override fun isLongPressDragEnabled(): Boolean {
+        return true
```
Comment (konempty) [inline app/src/main/kotlin/com/fakedevelopers/bidderbidder/ui/product_registration/DragAndDropCallback.kt]: 아래보다보니까 함수를 축약해서 많이 쓰시던데 여기도 fun is~~() = true로 통일하면 어떨까요??ㅋㅋㅋㅋ

**Review by konempty — APPROVED** [review]
고생하셨습니다 머지하셔도 좋을듯 합니다 💯

File: app/src/main/kotlin/com/fakedevelopers/bidderbidder/ui/product_registration/DragAndDropCallback.kt (hunk 2 above)
Comment (minseonglove) [reply app/src/main/kotlin/com/fakedevelopers/bidderbidder/ui/product_registration/DragAndDropCallback.kt]: ㄹㅇ

**Pull request merged**